include/engine/routing_algorithms/routing_base_mld.hpp

```diff
@@ -269,10 +270,29 @@ retrievePackedPathFromHeap(const SearchEngineData<Algorithm>::QueryHeap &forward
     return packed_path;
 }
 
-template <bool DIRECTION, typename Algorithm, typename... Args>
+template <typename Heap>
```

Comment:
My main goal here was to make all these algorithms to track distance if they are used in map matching and avoid any overhead in other places: I eventually found a way to do that via using `if constexpr`